docs/extends.md

```diff
@@ -337,24 +210,152 @@ with locally-defined values taking precedence:
       - BAR=local
       - BAZ=local
 
-Finally, for `volumes` and `devices`, Compose "merges" entries together with
-locally-defined bindings taking precedence:
 
-    # original service
-    volumes:
-      - /original-dir/foo:/foo
-      - /original-dir/bar:/bar
+## Multiple Compose files
 
-    # local service
-    volumes:
-      - /local-dir/bar:/bar
-      - /local-dir/baz/:baz
+Using multiple Compose files enables you to customize a composition for
+different environments or different workflows.
+
+### Understanding multiple Compose files
+
+Compose reads two files by default. A `docker-compose.yml` which should
+contain the base composition, and an optional `docker-compose.override.yml`,
+which may include overrides for existing services, or include new services.
+
+If a service is defined in both files, it will be merged using the same
+rules as the `extends` field (see
+[Adding and overriding configuration](#adding-and-overriding-configuration)),
+with one exception.  If a service contains `links` or `volumes_from` those
+fields will be copied over and replace any values in the original service,
+in the same way single-valued fields are copied.
+
+To use multiple override files, or an override file with a different name, you
+can use the `-f` option to specify the list of files. Files will merged in the
+order they're specified on the command line. See the
+[`docker-compose` command reference](./reference/docker-compose.md) for more
+information about using `-f`.
+
+When multiple files are used, all paths in the files are relative to the
+base Compose file (the first Compose file specified with `-f`). This is necessary
+because override files are not always valid Compose files, they may be just small
+fragments of configuration. Tracking which fragment of a service is relative to
+which path is difficult and confusing, so to keep paths easier to understand,
+all paths are kept relative to the base file.
+
+### Example use case
```

Comment:
No royal we's and everything happens in the now not future soooo...

--->

### Example use case

In this section are two common use cases for multiple compose files: changing a
composition for different environments, and running administrative tasks
against a composition.

#### Different environments

A common use case for multiple files is changing a development composition
for a production-like environment (which may be production, staging or CI).
To support these differences, you can split your Compose configuration into
a few different files:

Start with a base file that defines the canonical configuration for the
services.

**docker-compose.yml**

    web:
      image: example/my_web_app:latest
      links:
        - db
        - cache

    db:
      image: postgres:latest

    cache:
      image: redis:latest

The example development environment exposes some ports to the
host, mount our code as a volume, and build the web image.

**docker-compose.override.yml**


    web:
      build: .
      volumes:
        - '.:/code'
      ports:
        - 8883:80
      environment:
        DEBUG: 'true'

    db:
      command: '-d'
      ports:
        - 5432:5432

    cache:
      ports:
        - 6379:6379

When you run `docker-compose up` it reads the overrides automatically.

Now, it might be nicet to also use this composition in a production environment. So, create another override file (which might be stored in a different git
repo or managed by a different team).

**docker-compose.prod.yml**

    web:
      ports:
        - 80:80
      environment:
        PRODUCTION: 'true'

    cache:
      environment:
        TTL: '500'

To deploy with this production Compose file you can run

    docker-compose -f docker-compose.yml -f docker-compose.prod.yml up -d

This deploys all three services using the configuration in
`docker-compose.yml` and `docker-compose.prod.yml` (but not the
dev configuration in `docker-compose.override.yml`).

See [production](production.md) for more information about Compose in
production.

#### Administrative tasks

Another common use case is running adhoc or administrative tasks against one
or more services in a composition. This example demonstrates running a
database backup.

Start with a **docker-compose.yml**.

    web:
      image: example/my_web_app:latest
      links:
        - db

    db:
      image: postgres:latest

In a **docker-compose.admin.yml**  add a new service to run the database
export or backup.

    dbadmin:
      build: database_admin/
      links:
        - db


To start a normal environment run `docker-compose up -d`. To run a database
backup, use the `docker-compose.admin.yml` as well.

    docker-compose -f docker-compose.yml -f docker-compose.admin.yml \
        run dbadmin db-backup